brickschema/validate.py

```diff
@@ -0,0 +1,423 @@
+            self.__buildNamespaceDict(self.brickG)
+
+            # Remove rdfs:domain and rdfs:range.  The modified
+            # ontology will be used for pySHACL reasoning.
+            # See DESIGN.md for more discussion.
+
+            self.brickG.update('DELETE { ?s rdfs:domain ?o .} WHERE { ?s rdfs:domain ?o . }',
+                               initNs=self.namespaceDict)
+            self.brickG.update('DELETE { ?s rdfs:range ?o .} WHERE { ?s rdfs:range ?o . }',
+                               initNs=self.namespaceDict)
+
+        if useDefaultShapes:
+            data = pkgutil.get_data(__name__, "ontologies/BrickShape.ttl").decode()
+            self.shapeG.parse(source=io.StringIO(data), format='turtle')
+            self.__buildNamespaceDict(self.shapeG)
+
+
+    def validate(self, data_graph, shacl_graph=None, ont_graph=None,
+                 inference='rdfs', abort_on_error=False, advanced=True,
+                 meta_shacl=True, debug=False):
+        """
+        Validates data_graph against shacl_graph and ont_graph.
+
+        Args:
+            shacl_graph: default to BrickShape.ttl and shapes added with the addShape[Graph|File]() methods
+            ont_graph: default to Brick.ttl
+
+        Returns:
+            (tuple) (conforms,  result graph, result text)
+        """
+
+        self.log.info('wrapper function for pySHACL validate()')
+
+        sg = shacl_graph if shacl_graph else self.shapeG
+        og = ont_graph if ont_graph else self.brickG
+
+        self.data_graph = data_graph
+        (self.conforms, self.results_graph, self.results_text) = pyshacl.validate(
+            data_graph, shacl_graph=sg, ont_graph=og,
+            inference=inference, abort_on_error=abort_on_error,
+            meta_shacl=meta_shacl, debug=debug)
+
+        return (self.conforms, self.results_graph, self.results_text)
+
+
+    def addShapeFile(self, shapeFile):
+        """
+        Adds additional SHACL shape file into the existing shape graph.
+        """
+        self.log.info('load shape file %s' % shapeFile)
+        g = Graph()
+        g.parse(shapeFile, format='turtle')
+        self.shapeG = self.shapeG + g
+
+
+    def addShapeGraph(self, shapeGraph):
+        """
+        Adds additional SHACL shape graph into the existing shape graph.
+        """
+        self.log.info('load shape graph')
+        self.shapeG = self.shapeG + shapeGraph
+
+
+    def accumulatedNamespaces(self):
+        """
+        Convenient function to return the accmulated namepace dictionary.
+        """
+        return self.namespaceDict
+
+    def violationList(self):
+        """
+        Returns the violation graphs as a list. The
+        potential offending triples are in each violation graph.
+        """
+
+        if 'conforms' not in dir(self):
+            self.log.error('should call validate function first')
+            return None
+
+        self.__attachOffendingTriples()
+        return list(self.violationDict.values())
```

Comment:
How is this function meant to be used? Does it rely upon the execution of the `validate` method? If so, it may want to be part of the proposed `Result` object I discuss elsewhere